### DISCUSSION TITLE
Martin Source Endpoints Differences

### DISCUSSION THREAD

**Discussion by BigBoulard** [maplibre/martin#1238] [Q&A] [answered]
Hi Team,

Context: I'm trying to display a map of Belgium using MapLibre and Martin.

[In the doc](https://maplibre.org/martin/using.html) I can see this:
| URL | DESCRIPTION |
| -- | -- |
| /{sourceID} | Source TileJSON |
| /{sourceID}/{z}/{x}/{y} | Map Tiles |
| /{source1},…,{sourceN} | Composite Source TileJSON |
| /{source1},…,{sourceN}/{z}/{x}/{y} | Composite Source Tiles |

So, what's the difference between:

```json
"sources": {
    "osm-tiles": {
      "type": "vector",
      "tiles": [
        "https://mygateway:8080/planet_osm_point,planet_osm_line,planet_osm_roads,planet_osm_polygon"
      ],
      "minzoom": 0,
      "maxzoom": 22
    }
  },
```

or

```json
{
  "sources": {
    "osm-tiles": {
      "type": "vector",
      "tiles": [
        "https://mygateway:8080/planet_osm_line/{z}/{x}/{y}",
        "https://mygateway:8080/planet_osm_point/{z}/{x}/{y}",
        "https://mygateway:8080/planet_osm_polygon/{z}/{x}/{y}",
        "https://mygateway:8080/planet_osm_roads/{z}/{x}/{y}"
      ],
      "minzoom": 0,
      "maxzoom": 14
    }
  },
```

or

```json
{
  "sources": {
    "osm-tiles": {
      "type": "vector",
      "tiles": [
        "https://mygateway:8080/planet_osm_line",
        "https://mygateway:8080/planet_osm_point",
        "https://mygateway:8080/planet_osm_polygon",
        "https://mygateway:8080/planet_osm_roads"
      ],
      "minzoom": 0,
      "maxzoom": 14
    }
  },
```

Does it make a difference if Martin serves Tiles from a local `.mbtiles` file or a `Postgres` instance?

**Comment by nyurik** [comment, marked as answer]
Lots of good questions :)

* The comma-separated list means that you let Martin tile server do the tile merging - i.e. it will be just a single tile response, possibly cached by your CDN, rather than 4 individual responses.  There are positives and negatives with this approach - mostly having to do with cache optimization.
* Per [sources docs](https://maplibre.org/maplibre-style-spec/sources/), you can either give MapLibre GL one or more formatting strings with `"tiles": [...]` param that contains `{z}/{x}/{y}`, and also configure the rest of the tile source (min/max zoom,...), OR, you can give it a link to [tilejson](https://github.com/mapbox/tilejson-spec) using `"url"` -- in which case it must be just a single tilesource, and you don't need to set min/max zoom, ...

P.S. you will have to use a comma-separated format if you use `url` param with tilejson - because otherwise you need to create multiple tilesources, and configure them separately with different styling rules.

**Reply by BigBoulard** [reply]
Thank you so much. Cristal clear :)
